### PR TITLE
Cleanup bp common

### PR DESCRIPTION
**Pull request by black-parrot**

### CODE REVIEW

**Comment by black-parrot** [comment]
@dpetrisko

File: bp_common/bp_common_cfg_defines.vh [hunk 1]
```diff
@@ -0,0 +1,29 @@
+/**
+ *
+ * bp_common_cfg_defines.vh
+ *
+ */
+
+`ifndef BP_COMMON_CFG_DEFINES_VH
+`define BP_COMMON_CFG_DEFINES_VH
+
```
Comment (dpetrisko) [inline bp_common/bp_common_cfg_defines.vh]: comments

File: bp_common/bp_common_cfg_defines.vh [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+typedef logic[2:0] bp_mhartid_t;
+typedef logic[3:0] bp_lce_id_t;
+
+typedef struct packed
+{
+  bp_mhartid_t mhartid;
+  bp_lce_id_t  icache_id;
+  bp_lce_id_t  dcache_id;
+}  bp_proc_cfg_s;
+
```
Comment (dpetrisko) [inline bp_common/bp_common_cfg_defines.vh]: explain why we need a dynamic proc_cfg

File: bp_common/bp_common_fe_be_if.vh [hunk 3]
```diff
@@ -0,0 +1,326 @@
...
+ * ensure all exceptions are taken precisely.  The BE checks the PC in the
+ * instruction/PC pairs to make sure that the FE predictions correspond to
+ * correct architectural execution.
+ *
+ */
+
+`ifndef BP_COMMON_FE_BE_IF
+`define BP_COMMON_FE_BE_IF
+
+`include "bsg_defines.v"
```
Comment (dpetrisko) [inline bp_common/bp_common_fe_be_if.vh]: include in package

File: bp_common/bp_common_fe_be_if.vh [hunk 4]
```diff
@@ -0,0 +1,326 @@
...
+
+/*
+ * Clients need only use this macro to declare all parameterized structs for FE<->BE interface.
+ */
+`define declare_bp_common_fe_be_if_structs(vaddr_width_p,paddr_width_p,asid_width_p,branch_metadata_fwd_width_p) \
```
Comment (dpetrisko) [inline bp_common/bp_common_fe_be_if.vh]: _mp, spacing

File: bp_common/bp_common_fe_be_if.vh [hunk 5]
```diff
@@ -0,0 +1,326 @@
...
+     *                                                                                             \
+     * bp_fe_exception_s contains FE exception information.  Exceptions should be                  \
+     * serviced inline with instructions. Otherwise we have no way of knowing if this              \
+     * exception is eclipsed by a preceding branch mispredict.  FE does not receive                \
+     * interrupts, but may raise exceptions.                                                       \
+     */                                                                                            \
+    typedef struct packed {                                                                        \
+        logic [vaddr_width_p-1:0]                  vaddr;                                          \
+        bp_fe_exception_code_e                     exception_code;                                 \
+        logic [`bp_fe_exception_padding_width(vaddr_width_p,branch_metadata_fwd_width_p)-1:0] padding;                                        \
```
Comment (dpetrisko) [inline bp_common/bp_common_fe_be_if.vh]: alignment of padding

File: bp_common/bp_common_fe_be_if.vh [hunk 6]
```diff
@@ -0,0 +1,326 @@
...
+        logic [branch_metadata_fwd_width_p-1:0]   branch_metadata_fwd;                             \
+        logic [`bp_fe_fetch_padding_width(vaddr_width_p,branch_metadata_fwd_width_p)-1:0]    padding;                                         \
+    } bp_fe_fetch_s;                                                                               \
+                                                                                                   \
+    /*                                                                                             \
+     *                                                                                             \
+     * bp_fe_exception_s contains FE exception information.  Exceptions should be                  \
+     * serviced inline with instructions. Otherwise we have no way of knowing if this              \
+     * exception is eclipsed by a preceding branch mispredict.  FE does not receive                \
+     * interrupts, but may raise exceptions.                                                       \
```
Comment (dpetrisko) [inline bp_common/bp_common_fe_be_if.vh]: 2 spaces, not 4

File: bp_common/bp_common_fe_be_if.vh [hunk 7]
```diff
@@ -0,0 +1,326 @@
...
+    (`bp_fe_cmd_operands_u_width(vaddr_width_p,paddr_width_p,asid_width_p                          \
+                                 ,branch_metadata_fwd_width_p)
+
+`define bp_fe_cmd_itlb_map_width(vaddr_width_p,paddr_width_p,asid_width_p,branch_metadata_fwd_width_p) \
+    (`bp_fe_cmd_operands_u_width(vaddr_width_p,paddr_width_p,asid_width_p                          \
+                                 ,branch_metadata_fwd_width_p)
+
+`define bp_fe_pte_entry_leaf_width(paddr_width_p)                                                  \
+    (paddr_width_p+5*1)
+
```
Comment (dpetrisko) [inline bp_common/bp_common_fe_be_if.vh]: Fix backslash alignment issues

File: bp_common/bp_common_pkg.vh [hunk 8]
```diff
@@ -0,0 +1,21 @@
+
+    `include "bp_common_cfg_defines.vh"
```
Comment (dpetrisko) [inline bp_common/bp_common_pkg.vh]: include inside once people move to import bp_common_pkg vs include bp_common_if

File: bp_common/bp_common_me_if.vh [hunk 9]
```diff
@@ -0,0 +1,539 @@
...
+ *
+ * LCE to CCE Request
+ *
+ */
+
+/*
+ * bp_lce_cce_req_type_e specifies whether the containing message is related to a read or write
+ * cache miss request from and LCE.
+ */
+typedef enum logic {
```
Comment (dpetrisko) [inline bp_common/bp_common_me_if.vh]: enum spacing

**Pull request merged**